From 2ce201fadde89335ff6036c8b407846f82bcdbfd Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 12 Apr 2021 13:38:46 +0000
Subject: [PATCH] CMakeLists.txt: append to CMAKE_CXX_FLAGS instead of
 re-setting what is there

* with OE builds toolchain.cmake sets a lot of flags including the important
  --sysroot:
floam/0.1.0-1-r0/git$ grep CMAKE_CXX_FLAGS ../toolchain.cmake
set( CMAKE_CXX_FLAGS " -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse  --sysroot=/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/jenkins/mjansa/build/ros/ros
1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0=/usr/src/debug/floam/0.1.0-1-r0                      -fdebug-prefix-map=/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0=/usr/src/debug/floam/0.1.0-1-r0
 -fdebug-prefix-map=/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot=                      -fdebug-prefix-map=/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot-nati
ve=  -fvisibility-inlines-hidden  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse  --sysroot=/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot" CACHE STRING "CXXFLAGS" )
set( CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG" CACHE STRING "Additional CXXFLAGS for release" )

* fixes:
...
FAILED: CMakeFiles/floam_mapping.dir/src/lidar_mapping.cpp.o
/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot-native/usr/bin/x86_64-oe-linux/x86_64-oe-linux-g++ -DCERES_EXPORT_INTERNAL_SYMBOLS -DGOOGLE_GLOG_DLL_DECL="" -DGOOGLE_GLOG_DLL_DECL_FOR_UNITTESTS="" -DROSCONSOLE_BACKEND_LOG4CXX -DROS_BUILD_SHARED_LIBS=1 -DROS_PACKAGE_NAME=\"floam\" -Dfloam_mapping_EXPORTS -I/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/git/include -I/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/opt/ros/noetic/include -I/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/opt/ros/noetic/share/xmlrpcpp/cmake/../../../include/xmlrpcpp -I/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/usr/include/eigen3 -I/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/usr/include/pcl-1.8 -std=c++14 -O3 -Wall -g -fPIC -MD -MT CMakeFiles/floam_mapping.dir/src/lidar_mapping.cpp.o -MF CMakeFiles/floam_mapping.dir/src/lidar_mapping.cpp.o.d -o CMakeFiles/floam_mapping.dir/src/lidar_mapping.cpp.o -c /jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/git/src/lidar_mapping.cpp
In file included from /jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/usr/include/eigen3/Eigen/StdVector:14,
                 from /jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/usr/include/pcl-1.8/pcl/point_cloud.h:46,
                 from /jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/git/include/floam/lidar_mapping.hpp:11,
                 from /jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/git/src/lidar_mapping.cpp:7:
/jenkins/mjansa/build/ros/ros1-noetic-hardknott/tmp-glibc/work/core2-64-oe-linux/floam/0.1.0-1-r0/recipe-sysroot/usr/include/eigen3/Eigen/Core:82:12: fatal error: new: No such file or directory
   82 |   #include <new>
      |            ^~~~~
compilation terminated.

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e85a7d5..0769791 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ project(floam)
 set(CMAKE_BUILD_TYPE "Release")
 
 # PCL requires C++ 14 and above
-set(CMAKE_CXX_FLAGS "-std=c++14")
-set(CMAKE_CXX_FLAGS_RELEASE "-O3 -Wall -g")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -Wall -g")
 
 find_package(catkin REQUIRED COMPONENTS
   geometry_msgs
